src/aws/cis-1.4.0/rules/aws-cis-1.4.0-1.7.ts

```diff
@@ -0,0 +1,70 @@
+// AWS CIS 1.2.0 Rule equivalent 1.1
+export default {
+  id: 'aws-cis-1.4.0-1.7',  
+  title: 'AWS CIS 1.7 Eliminate use of the \'root\' user for administrative and daily tasks',
+  
+  description: 'With the creation of an AWS account, a \'root user\' is created that cannot be disabled or deleted. That user has unrestricted access to and control over all resources in the AWS account. It is highly recommended that the use of this account be avoided for everyday tasks.',
+  
+  audit: `**From Console:**
+  
+  1. Login to the AWS Management Console at https://console.aws.amazon.com/iam/
+  2. In the left pane, click *Credential Report*
+  3. Click on *Download Report*
+  4. Open of Save the file locally
+  5. Locate the *<root account>* under the user column
+  6. Review *password_last_used*, *access_key_1_last_used_date*, *access_key_2_last_used_date* to determine when the 'root user' was last used.
+  
+  **From Command Line:**
+  Run the following CLI commands to provide a credential report for determining the last time the 'root user' was used:
+  
+      aws iam generate-credential-report
+  
+      aws iam get-credential-report --query 'Content' --output text | base64 -d | cut -d, -f1,5,11,16 | grep -B1 '<root_account>'
+  
+  Review *password_last_used*, *access_key_1_last_used_date*, *access_key_2_last_used_date* to determine when the root user was last used.
+  
+  **Note:** There are a few conditions under which the use of the 'root' user account is required. Please see the reference links for all of the tasks that require use of the 'root' user.`,
+  
+  rationale: 'The \'root user\' has unrestricted access to and control over all account resources. Use of it is inconsistent with the principles of least privilege and separation of duties, and can lead to unnecessary harm due to error or account compromise.',
+  
+  remediation: `Remediation:
+  
+  If you find that the 'root' user account is being used for daily activity to include administrative tasks that do not require the 'root' user:
+  
+  1. Change the 'root' user password.
+  2. Deactivate or delete any access keys associate with the 'root' user.
+  
+  **Remember, anyone who has 'root' user credentials for your AWS account has unrestricted access to and control of all the resources in your account, including billing information.`,
+  
+  references: [
+      'https://docs.aws.amazon.com/IAM/latest/UserGuide/best-practices.html',
+      'https://docs.aws.amazon.com/IAM/latest/UserGuide/id_root-user.html',
+      'https://docs.aws.amazon.com/general/latest/gr/aws_tasks-that-require-root.html',
+  ],
+  gql: `{
+    queryawsIamUser(filter: { name: { eq: "root" } }) {
+      id
+      arn
+      accountId
+       __typename
+      passwordLastUsed
+      passwordEnabled
+    }
+  }`,
+  resource: 'queryawsIamUser[*]',
+  severity: 'high',
+  conditions: {
```

Comment:
reading the description, do we also need to check the access key last used here?

Reply:
or check that it has no access keys

Reply:
This part of the sql query looks to check against access keys, we should probably do the same (check access keys and password)
```
when access_key_1_last_used_date <= (current_date - interval '90' day)  then 'alarm'
    when access_key_2_last_used_date <= (current_date - interval '90' day)  then 'alarm'
```

Reply:
@tyler-dunkel  yes, I saw that too and it is correct. But also that is what is being done in the rule, we are validating the password and both access keys. We return both (key_1 and key_2) inside the accessKeyData array. Unless I'm missing something.

Reply:
I might be missing something. I only see the password enabled part of the rule conditions here

Reply:
nvm I see the changes now